app/forms/state_file/id_grocery_credit_form.rb

```diff
@@ -0,0 +1,86 @@
+module StateFile
+  class IdGroceryCreditForm < QuestionsForm
+    attr_accessor :dependents_attributes
+    delegate :dependents, to: :intake
+
+    set_attributes_for(
+      :intake,
+      :household_has_grocery_credit_ineligible_months,
+      :primary_has_grocery_credit_ineligible_months,
+      :spouse_has_grocery_credit_ineligible_months,
+      :primary_months_ineligible_for_grocery_credit,
+      :spouse_months_ineligible_for_grocery_credit
+    )
+
+    validates :household_has_grocery_credit_ineligible_months, inclusion: { in: %w[yes no], message: :blank }
+
+    validates :primary_months_ineligible_for_grocery_credit,
+              numericality: {
+                greater_than_or_equal_to: 0,
+                message: :blank
+              },
+              if: -> { primary_has_grocery_credit_ineligible_months == "yes" }
+    validates :spouse_months_ineligible_for_grocery_credit,
+              numericality: {
+                greater_than_or_equal_to: 0,
+                message: :blank
+              },
+              if: -> { spouse_has_grocery_credit_ineligible_months == "yes" }
+
+    def initialize(intake = nil, params = nil)
+      super
+      if params.present?
+        @intake.assign_attributes(dependents_attributes: dependents_attributes.to_h)
+      end
+    end
+
+    def save
+      @intake.update!(attributes_to_save)
+    end
+
+    def attributes_to_save
+      base_attrs = attributes_for(:intake)
+      updated_dependent_data = dependents_attributes
+
+      # set household member "has months" answers to no if household "has months" answer is no
+      if base_attrs[:household_has_grocery_credit_ineligible_months] == 'no'
+        base_attrs[:primary_has_grocery_credit_ineligible_months] = 'no'
+        base_attrs[:spouse_has_grocery_credit_ineligible_months] = 'no'
+        updated_dependent_data = updated_dependent_data.to_h do |k, v|
+          if v.key?(:id_has_grocery_credit_ineligible_months)
+            [k, v.merge(id_has_grocery_credit_ineligible_months: 'no')]
+          else
+            [k, v]
+          end
+        end
+      end
+
+      # set household months to nil/0 if their "has months" answer is no
+      if base_attrs[:primary_has_grocery_credit_ineligible_months] == 'no'
+        base_attrs[:primary_months_ineligible_for_grocery_credit] = ''
+      end
+      if base_attrs[:spouse_has_grocery_credit_ineligible_months] == 'no'
+        base_attrs[:spouse_months_ineligible_for_grocery_credit] = ''
+      end
+
+      updated_dependent_data = updated_dependent_data.to_h do |k, v|
+        credit_count_key = v.key?(:id_months_ineligible_for_grocery_credit)
+        has_matching_no = updated_dependent_data.any? do |_, v2|
+          v[:id] == v2[:id] && v2[:id_has_grocery_credit_ineligible_months] == 'no'
+        end
```

Comment:
why is this going through the `updated_dependents_data` again to find the matching `v`? Isn't this the same as:

```
has_matching_no = v[:id_has_grocery_credit_ineligible_months] == 'no'
```
can we simplify this whole section to
```
      updated_dependent_data = updated_dependent_data.to_h do |k, v|
         if v.key?(:id_months_ineligible_for_grocery_credit) && v[:id_has_grocery_credit_ineligible_months] == 'no'
            [k, v.merge(id_months_ineligible_for_grocery_credit: '')]
         else
            [k, v]

      end
```

Reply:
It's because the `dependent_params` hash has separate entries for the `id_has_grocery_credit_ineligible_months` and `id_months_ineligible_for_grocery_credit` params. I believe that this is because they're in separate `fields_for :dependents`, which generate their own param "sub-hashes", but which can be corresponded by the database IDs of the dependents (as we do here). Looking in the same place for both keys won't work unless we do a prior step of merging the params together so that each dependent has just one entry. I felt like the complexity was about the same either way, so Hugo and I went with this approach.

Reply:
Oh yeah! Forgot about that piece 🤪